src/libraries/System.Linq/tests/ElementAtTests.cs

```diff
@@ -229,14 +229,14 @@ public void NonEmptySource_Consistency_ThrowsIListIndexerException()
             Assert.Throws<ArgumentOutOfRangeException>("index", () => source.ElementAt(^11));
             // ImmutableArray<T> implements IList<T>. ElementAt calls ImmutableArray<T>'s indexer, which throws IndexOutOfRangeException instead of ArgumentOutOfRangeException.
             Assert.Throws<IndexOutOfRangeException>(() => ImmutableArray.Create(source).ElementAt(-1));
-            Assert.Throws<IndexOutOfRangeException>(() => ImmutableArray.Create(source).ElementAt(^11));
+            Assert.Throws<ArgumentOutOfRangeException>(() => ImmutableArray.Create(source).ElementAt(^11));
```

Comment:
Seems the original exception type is more appropriate in this context?

Reply:
It's unfortunate that we get different exception types for ImmutableArray here than we do with arrays, lists, etc. That's what I was trying to unify. But, it seems we have even more tests validating this exact exception type with ElementAt and ImmutableArray; rather than trying to fix them all, I'll revert the part that changed this.